File: apps/waiter/src/services/table.service.ts
import api from './api';

export interface Table {
  id: string;
  number: number;
  name?: string;
  capacity: number;
  status: 'FREE' | 'OCCUPIED' | 'RESERVED' | 'CLEANING';
  isActive: boolean;
  activeOrders?: { id: string }[];
  orders?: { id: string; status: string }[];
}

interface TablesResponse {
  data: Table[];
}

interface TableResponse {
  data: Table;
}

export const tableService = {
  getAll: async (): Promise<TablesResponse> => {
    const { data: response } = await api.get('/tables');
    // API returns { success, data: [...tables] }
    const tables: Table[] = (response.data || []).map((t: Table & { orders?: { id: string; status: string }[] }) => ({
      ...t,
      activeOrders: t.orders?.filter((o) =>
        ['NEW', 'CONFIRMED', 'PREPARING', 'READY'].includes(o.status)
      ) || [],
    }));
    return { data: tables };
  },

  getById: async (id: string): Promise<TableResponse> => {
    const { data: response } = await api.get(`/tables/${id}`);
    return { data: response.data };
  },

  updateStatus: async (id: string, status: Table['status']): Promise<void> => {
    await api.patch(`/tables/${id}/status`, { status });
  },
};

File: apps/waiter_flutter/lib/services/table_service.dart
import '../core/api.dart';
import '../models/table.dart';

class TableService {
  Future<List<TableModel>> getAll() async {
    final response = await dio.get('/tables');
    final data = response.data;
    final Map<String, dynamic> body = data is Map<String, dynamic> ? data : {};
    final List<dynamic> list = body['data'] as List<dynamic>? ??
        body['tables'] as List<dynamic>? ??
        (data is List ? data : []);
    return list.map((e) => TableModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<TableModel> getById(String id) async {
    final response = await dio.get('/tables/$id');
    final data = response.data;
    final Map<String, dynamic> body = data is Map<String, dynamic> ? data : {};
    final tableData = body['data'] ?? body;
    return TableModel.fromJson(tableData as Map<String, dynamic>);
  }
}
